env0/resource_configuration_variable.go

```diff
@@ -139,7 +147,11 @@ func whichScope(d *schema.ResourceData) (client.Scope, string) {
 		scopeId = projectId.(string)
 	}
 
-	if templateId, ok := d.GetOk("template_id"); ok {
+	if subEnvironmentAlias, ok := d.GetOk("sub_environment_alias"); ok {
```

Comment:
first check for alias, only if there's no alias, check templateId (if else is needed).